math/mathcore/inc/Math/Util.h

```diff
@@ -262,31 +262,62 @@ namespace ROOT {
          return *this;
        }
 
-       /// Add `arg` into accumulator. Does not vectorise.
+       /// Add other KahanSum into accumulator. Does not vectorise.
+       ///
+       /// Based on KahanIncrement from:
+       /// Y. Tian, S. Tatikonda and B. Reinwald, "Scalable and Numerically Stable Descriptive Statistics in SystemML," 2012 IEEE 28th International Conference on Data Engineering, 2012, pp. 1351-1359, doi: 10.1109/ICDE.2012.12.
+       /// Note that while Tian et al. add the carry in the first step, we subtract
+       /// the carry, in accordance with the Add(Indexed) implementation(s) above.
+       /// This is purely an implementation choice that has no impact on performance.
```

Comment:
I implemented everything only for non-vectorized KahanSums for the time being, since those are the only ones using these operators. At least, they were already non-vectorized before this change, so that's why I assumed I needn't implement this for the vectorized case.

Reply:
ok, then maybe let's add a comment in the code to make it clear that the += and -= operator will work only for other with N=1. The problem is now you can add a KahanSum<T,4> withKahanSum<T,1>, so we should add also a static assert for
N==1.

Reply:
Ah that's a very good point, I will add the assert. The comment was already there btw ("Does not vectorize."), I kept that in.

Reply:
Hm, actually, thinking about it more, I think different N's are not really a problem. The code will work, it just won't use the other `N-1` array slots and won't vectorize, but that's algebraically not an issue. I will try to add a test to show this.

Reply:
I added this in the next commit, please have a look :)